pkg/cluster/driver/olvm/driver.go

```diff
@@ -0,0 +1,241 @@
+	haveError := logutils.WaitFor(logutils.Info, []*logutils.Waiter{
+		{
+			Message: "Waiting for Core Cluster API Controllers",
+			WaitFunction: func(i interface{}) error {
+				return k8s.WaitForDeployment(kubeClient, constants.CoreCAPINamespace, constants.CoreCAPIDeployment, 1)
+			},
+		},
+		{
+			Message: "Waiting for Kubadm Boostrap Cluster API Controllers",
+			WaitFunction: func(i interface{}) error {
+				return k8s.WaitForDeployment(kubeClient, constants.KubeadmBootstrapCAPINamespace, constants.KubeadmBootstrapCAPIDeployment, 1)
+			},
+		},
+		{
+			Message: "Waiting for Kubadm Control Plane Cluster API Controllers",
+			WaitFunction: func(i interface{}) error {
+				return k8s.WaitForDeployment(kubeClient, constants.KubeadmControlPlaneCAPINamespace, constants.KubeadmControlPlaneCAPIDeployment, 1)
+			},
+		},
+		//{
+		//	Message: "Waiting for Olvm Cluster API Controllers",
+		//	WaitFunction: func(i interface{}) error {
+		//		return k8s.WaitForDeployment(kubeClient, constants.OLVMCAPIOperatorNamespace, constants.OLVMCAPIDeployment, 1)
+		//	},
+		//},
+	})
+	if haveError {
+		return fmt.Errorf("Not all Cluster API controllers became available")
+	}
+	return nil
+}
+
+// getClusterObject gets the CAPI cluster object that is being started or deleted.
+func (cad *OlvmDriver) getClusterObject() (unstructured.Unstructured, error) {
+	clusterObj, err := k8s.FindIn(cad.ClusterResources, func(u unstructured.Unstructured) bool {
+		if u.GetKind() != "Cluster" {
+			return false
+		}
+		if u.GetAPIVersion() != "cluster.x-k8s.io/v1beta1" {
+			return false
+		}
+		_, ok := u.GetLabels()[ClusterNameLabel]
+		return ok
+	})
+	if err != nil {
+		if k8s.IsNotExist(err) {
+			return unstructured.Unstructured{}, fmt.Errorf("Cluster resources do not include a valid cluster.x-k8s.io/v1beta1/Cluster")
+		} else {
+			return unstructured.Unstructured{}, err
+		}
+	}
+	return clusterObj, err
+}
+
+// applyResources creates resources in a cluster if the resource does not
+// already exist.  If the resource already exists, it is not modified.
+func (cad *OlvmDriver) applyResources(restConfig *rest.Config) error {
+	resources, err := k8s.Unmarshall(bufio.NewReader(bytes.NewBufferString(cad.ClusterResources)))
+	if err != nil {
+		return err
+	}
+
+	for _, r := range resources {
+		err = k8s.CreateResourceIfNotExist(restConfig, &r)
+		if err != nil && !strings.Contains(err.Error(), "not found") {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (cad *OlvmDriver) Join(kubeconfigPath string, controlPlaneNodes int, workerNodes int) error {
+	return fmt.Errorf("Joining new nodes to this cluster is done by editing the KubeadmControlPlane and MachineDeployment resources in the management cluster")
+}
+
+func (cad *OlvmDriver) Stop() error {
+	return fmt.Errorf("OlvmDriver.Stop() is not implemented")
+}
+
+func (cad *OlvmDriver) GetKubeconfigPath() string {
+	return cad.KubeConfig
+}
+
+func (cad *OlvmDriver) GetKubeAPIServerAddress() string {
```

Comment:
The kube-proxy stuff I'm doing now relies on this value.  It should point to the IP of the Kubernetes API Server (virtualIp or loadBalancer).

Reply:
fixed, returns http://<ip>:<port>